gateway/ln-gateway/src/lib.rs

```diff
@@ -329,14 +375,75 @@ impl Gateway {
         &self,
         BackupPayload { federation_id }: BackupPayload,
     ) -> Result<()> {
-        self.select_actor(federation_id).await?.backup().await
+        self.select_actor(federation_id)
+            .await?
+            .read()
+            .await
+            .backup()
+            .await
     }
 
     async fn handle_restore_msg(
         &self,
         RestorePayload { federation_id }: RestorePayload,
     ) -> Result<()> {
-        self.select_actor(federation_id).await?.restore().await
+        self.select_actor(federation_id)
+            .await?
+            .read()
+            .await
+            .restore()
+            .await
+    }
+
+    async fn handle_lightning_reconnect(
+        &mut self,
+        payload: LightningReconnectPayload,
+    ) -> Result<()> {
+        let LightningReconnectPayload { node_type } = payload;
+
+        let actors = self.actors.lock().await;
+
+        // Stop all threads that are listening for HTLCs
+        tracing::info!("Stopping all HTLC subscription threads.");
+        for (_, actor) in actors.iter() {
+            actor.write().await.stop_subscribing_htlcs().await?;
+        }
+
+        // Disconnect the lightning connection, then reconnect it
+        self.lnrpc.write().await.disconnect().await?;
```

Comment:
Make an issue to see if we can get rid of `disconnect` and `connect`

Reply:
Made issue, will work on this next: https://github.com/fedimint/fedimint/issues/2069